<a href="https://colab.research.google.com/github/AnniMV/JODA2021/blob/main/Oppimispk.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Viikko 1
Tästä alkaa "Johdanto datatieteeseen"-kussin oppimispäiväkirja. Viikko alkoi pehmeästi johdatuksella datatieteeseen ja kurssin sisältöön ja sen suorittamiseen, ja ensivaikutelmani on, että tämä on ensimmäinen ja käytännöllisin yliopiston kurssi koko opiskeluhistoriani ajalta, jossa teorian lisäksi tulen oppimaan käytännössä eri työkalujen käyttöä, ja jolla on yhteyksiä myös "todelliseen maailmaan". Odotukset ovat siis korkealla kurssin suhteen, ja toivon oppivani paljon datan keräämisestä ja visualisoinnista.

Kurssi alkoi kurssin suorittamisen läpikäymisellä ja siitä mitä datatiede pitää sisällään. Datatieteeseen ja data-analyytikon ominaisuuksiin kuuluu monitahoisesti useampi osa-alue, joita hallita; matematiikka ja tilastotiede, ohjelmointi- ja tietotekniset taidot, kommunikointi- ja visualisointitaidot sekä liiketoiminta- ja ongelmanratkaisutaidot. Sen sijaan että yksi henkilö hallitsisi kaikki osa-alueet laaja-alaisesti, voidaan haluttu päämäärä saavuttaa myös kokoamalla sopivasti eri kompetenssit omaava henkilöiden ryhmä, jotta käytetty tieto saadaan koottua ja esitettyä päätöksen tekoa tukevaksi. Data-analytiikassa käytetään paljon Python ohjelmointikieltä, ja Jupyter Notebook on kätevä tapa esittää Python koodia ja kommentteja. Tutustuttiin Pythonin Pandas data-analytiikkatyökaluun.
- Hyvä data-analyytikko hallitsee useampia osa-alueita Itsellä on ainakin osasta jo tietämystä.
- Dataa on valtavasti ja kasvavassa määrin tarjolla.
- Datan kerääminen ja käsittely vaatii työmäärältään eniten eri analyysivaiheista.
- Datan esittämisellä voidaan myös ohjailla ihmisten ajattelutapaa korostamalla tiettyjä osia ja jättämällä toisia osia pois.
- Opettelen käyttämään Googlen Colabia, sen käytön etuina on tallennustilan ja laskentakapasiteetin säästäminen omalta koneelta, ja helppo tallentaminen GitHubiin. 


Tilastotiede on yksi osa-alue josta data-analyytikolla pitäisi olla tietämystä. Linkissä https://seeing-theory.brown.edu/bayesian-inference/index.html voi testata itse bayesilaista päättelyä esimerkin kautta.

Osallistuin viikon opetukseen, ja aineistona käytin tiedostoa https://github.com/jodatut/2020/raw/master/koodiesimerkit/SalesJan2009.csv 

Koodissa tarkastellaan miten paljon eri tuotteita (Product) on tilattu eri maihin.




In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

url = 'https://github.com/jodatut/2020/raw/master/koodiesimerkit/SalesJan2009.csv'
df = pd.read_csv(url)
print('List of attributes:', df.columns.values.tolist())
df.head()

List of attributes: ['Transaction_date', 'Product', 'Price', 'Payment_Type', 'Name', 'City', 'State', 'Country', 'Account_Created', 'Last_Login', 'Latitude', 'Longitude']


Transaction_date   Product  Price  ...    Last_Login   Latitude   Longitude
0      1/2/09 6:17  Product1   1200  ...   1/2/09 6:08  51.500000   -1.116667
1      1/2/09 4:53  Product1   1200  ...   1/2/09 7:49  39.195000  -94.681940
2     1/2/09 13:08  Product1   1200  ...  1/3/09 12:32  46.188060 -123.830000
3     1/3/09 14:44  Product1   1200  ...  1/3/09 14:22 -36.133333  144.750000
4     1/4/09 12:56  Product2   3600  ...  1/4/09 12:45  33.520560  -86.802500

[5 rows x 12 columns]

Tarkastellaan, kuinka monta kappaletta kutakin tuotetta on myyty maittain ja yhteensä.

In [ ]:
products = df.loc[:,["Country", "Product"]]
products.head()
dfprod = products.groupby(["Country"])["Product"].value_counts()
dfprod



Country         Product  
Argentina       Product1       1
Australia       Product1      30
                Product2       8
Austria         Product1       6
                Product2       1
                            ... 
United Kingdom  Product2      10
United States   Product1     401
                Product2      55
                Product3       7
                Product3       1
Name: Product, Length: 83, dtype: int64

In [ ]:

print(products.Product.unique())

number_of_prod = products["Product"].value_counts()
print(number_of_prod)

['Product1' 'Product2' 'Product3' 'Product3 ']
Product1     849
Product2     136
Product3      14
Product3       1
Name: Product, dtype: int64


# Viikko 2


Toisen viikon aiheena on datan kerääminen ja jalostaminen.
Datatieteen prosessimalli sisältää vaiheita. Datan keräämisvaiheessa on päätettävä mitä datalähdettä käytetään ja miten data kerätään. Datasta valitaan ydin jolla voidaan saavuttaa vastaus asetettuun kysymykseen tai ongelmaan. Analyysivaiheessa kirjoitetaan koodia jolla dataa käsitellään, tarkistetaan tuloksia. Arviointivaiheessa vertaillaan tuloksia, ja mietitään vastaako tulokset asetettuun kysymykseen, ja mietitään tarvitseeko tehdä muutoksia. Viimeiseksi, tulokset otetaan käyttöön, jotta analyysin teosta olisi hyötyä. Tulokset julkaistaan ja tulevat toimenpiteet voidaan toteuttaa ainakin osittain analyysiin nojautuen.

Liiketoiminnassa analytiikkaa voidaan hyödyntää eri tasoilla; esimerkiksi seurata liiketoiminnan perustietoja ja sen kehittymistä tai käyttää ennustamaan tulevaisuuden tapahtumia. Vaikka data-analytiikka liiketoiminnan apuna on ollut puheenaihe pidempään, kaikki yritykset eivät vielä hyödynnä sitä. Puutteita voi olla datan keräämisessä, sen tallentamisessa, teknologioiden hallitsemisessa ja analysointivaiheen toteuttamisessa.

Dataa voidaan kerätä verkkosivuilta ryömijöillä ja raapijoilla. Ryömijä on automatisoitu ohjelma, joka käy läpi verkkosivuja indeksoimalla haettavaa dataa, raapijalla kerätään automatisoidusti määrättyä tietoa verkkosivuilta. Käytännön toteutukset ovat usein näiden yhdistelmiä. Tälläkin tavoin kerätyn data-aineiston käytön laillisuus ja eettisyyskysymykset on huomioitava.

Ennen kuin datan analysointi varsinaisesti alkaa, sitä valmistellaan ja jalostetaan käytettävämpään muotoon. Datatyypit määrittävät mitä operaatioita datalle voi tehdä. Datasta voidaan tutkia esimerkiksi, ovatko datan tyypit tarkoituksenmukaista muotoa, ovatko tyhjät arvot merkitseviä analyysin kannalta ja miten tyhjät arvot voisi tulkita.

- Data-analytiikan käyttöaste yrityksessä riippuu varmasti paljon yrityksen koosta, sillä siitä riippuu, kuinka paljon yrityksellä on varaa panostaa datan käsittelyyn ja osaavaan henkilökuntaan. Voisi väittää että pienimmätkin yritykset tekevät vähintään perustason raportointia yrityksestään, mutta varsinaisesta data-analytiikasta kaikki eivät tule välttämättä taloudellisesti hyötymään.
- Ryömijöillä ja raapijoilla saadaan kerättyä dataa verkkosivuilta.
- Kun ryömijöitä ja raapijoita käytetään, on otettava huomioon käyttöehdot ja kohteen määrittelemät rajoitukset, jotta oma toiminta pysyy laillisuuden rajoissa.
- Kun käytettävä aineisto on kerätty ja koottu yhteen, dataa tutkitaan ja jalostetaan käytettävämpään muotoon.
- Dataa tutkimalla ja tarkastamalla voidaan varmistaa että datan tyypit ja sisältö vastaavat käyttötarkoitusta.


Toteutetaan  garnstudio.com raapija käyttäen pythonin scrapy kirjastoa. Haetaan garnstudion myymien lankojen nimet ja niiden hinnat.

In [ ]:
!pip install scrapy

     |████████████████████████████████| 245kB 8.5MB/s 
     |████████████████████████████████| 3.1MB 15.4MB/s 
     |████████████████████████████████| 256kB 39.3MB/s 
     |████████████████████████████████| 3.2MB 49.0MB/s 
     |████████████████████████████████| 61kB 6.8MB/s 
     |████████████████████████████████| 3.2MB 50.8MB/s 
     |████████████████████████████████| 81kB 7.7MB/s 
  Created wheel for protego: filename=Protego-0.1.16-cp37-none-any.whl size=7766 sha256=a236d68a5aa3e15698bcfaa8ad165fb799c78a35b020ad5fe2953742f1419a5d
  Stored in directory: /root/.cache/pip/wheels/51/01/d1/4a2286a976dccd025ba679acacfe37320540df0f2283ecab12
  Created wheel for PyDispatcher: filename=PyDispatcher-2.0.5-cp37-none-any.whl size=11517 sha256=b10bd68e9eac716163c5b2e8920a1152c4e4705d9f5f40b73be64c01f9f3e9f5
  Stored in directory: /root/.cache/pip/wheels/88/99/96/cfef6665f9cb1522ee6757ae5955feedf2fe25f1737f91fa7f
Successfully built protego PyDispatcher


In [ ]:
!scrapy genspider garnstudio_scraper garnstudio.com

Created spider 'garnstudio_scraper' using template 'basic' 


Raapija on seuraavanlainen:

In [ ]:
import scrapy


class GarnstudioScraperSpider(scrapy.Spider):
    name = 'garnstudio_scraper'
    allowed_domains = ['garnstudio.com']
    start_urls = ['https://www.garnstudio.com/yarns.php?cid=11']

    def parse(self, response):
        yarns = response.css('.prod_desc > h3').css('a::text').extract()
        prices = response.css('.price > p').css('strong::text').extract()
        for i in range(len(yarns)):
          yarn_prices = {
              'yarn' : yarns[i],
              'price' : prices[i]
          }         
          yield yarn_prices


In [ ]:
!scrapy runspider garnstudio_scraper.py -o out.json

2021-04-04 20:33:40 [scrapy.utils.log] INFO: Scrapy 2.4.1 started (bot: scrapybot)
2021-04-04 20:33:40 [scrapy.utils.log] INFO: Versions: lxml 4.2.6.0, libxml2 2.9.8, cssselect 1.1.0, parsel 1.6.0, w3lib 1.22.0, Twisted 21.2.0, Python 3.7.10 (default, Feb 20 2021, 21:17:23) - [GCC 7.5.0], pyOpenSSL 20.0.1 (OpenSSL 1.1.1k  25 Mar 2021), cryptography 3.4.7, Platform Linux-4.19.112+-x86_64-with-Ubuntu-18.04-bionic
2021-04-04 20:33:40 [scrapy.utils.log] DEBUG: Using reactor: twisted.internet.epollreactor.EPollReactor
2021-04-04 20:33:40 [scrapy.crawler] INFO: Overridden settings:
{'SPIDER_LOADER_WARN_ONLY': True}
2021-04-04 20:33:40 [scrapy.extensions.telnet] INFO: Telnet Password: 7821b237bf11107f
2021-04-04 20:33:40 [scrapy.middleware] INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats']
2021-04-04 20:33

Haetut tiedot ovat out.json-tiedostossa.

[
{"yarn": "DROPS Air", "price": "5.70 EUR"},
{"yarn": "DROPS Alaska", "price": "2.40 EUR"},
{"yarn": "DROPS Alpaca", "price": "4.00 EUR"},
{"yarn": "DROPS Alpaca Boucl\u00e9", "price": "3.80 EUR"},
{"yarn": "DROPS Andes", "price": "5.40 EUR"},
{"yarn": "DROPS Baby Merino", "price": "3.80 EUR"},
{"yarn": "DROPS BabyAlpaca Silk", "price": "6.10 EUR"},
{"yarn": "DROPS Belle", "price": "2.40 EUR"},
{"yarn": "DROPS Big Delight", "price": "7.10 EUR"},
{"yarn": "DROPS Big Merino", "price": "3.80 EUR"},
{"yarn": "DROPS Bomull-Lin", "price": "3.10 EUR"},
{"yarn": "DROPS Brushed Alpaca Silk", "price": "3.10 EUR"},
{"yarn": "DROPS Cotton Light", "price": "1.50 EUR"},
{"yarn": "DROPS Cotton Merino", "price": "4.00 EUR"},
{"yarn": "DROPS Delight", "price": "3.60 EUR"},
{"yarn": "DROPS Fabel", "price": "2.70 EUR"},
{"yarn": "DROPS Flora", "price": "2.70 EUR"},
{"yarn": "DROPS Glitter", "price": "2.20 EUR"},
{"yarn": "DROPS Karisma", "price": "2.70 EUR"},
{"yarn": "DROPS Kid-Silk", "price": "4.95 EUR"},
{"yarn": "DROPS Lima", "price": "2.70 EUR"},
{"yarn": "DROPS Loves You 7", "price": "1.10 EUR"},
{"yarn": "DROPS Loves You 8", "price": "1.10 EUR"},
{"yarn": "DROPS Loves You 9", "price": "0.90 EUR"},
{"yarn": "DROPS Melody", "price": "5.30 EUR"},
{"yarn": "DROPS Merino Extra Fine", "price": "3.80 EUR"},
{"yarn": "DROPS Muskat", "price": "2.30 EUR"},
{"yarn": "DROPS Nepal", "price": "2.80 EUR"},
{"yarn": "DROPS Nord", "price": "2.80 EUR"},
{"yarn": "DROPS Paris", "price": "1.35 EUR"},
{"yarn": "DROPS Polaris", "price": "5.10 EUR"},
{"yarn": "DROPS Puna", "price": "3.40 EUR"},
{"yarn": "DROPS Safran", "price": "1.70 EUR"},
{"yarn": "DROPS Sky", "price": "5.90 EUR"},
{"yarn": "DROPS Snow", "price": "2.40 EUR"},
{"yarn": "DROPS Wish", "price": "5.70 EUR"}


# Viikko 3

Koneoppiminen on tekoälyn osa-alue ja se voidaan jaotella ohjattuun, ohjaamattomaan ja vahvistettuun oppimiseen. 

Ohjatulla oppimisella tarkoitetaan tilannetta, jossa koneelle opetetaan opetusdatalla ennustamaan jotakin tapahtumaa. Usein data on jaettu kahteen osaan niin että testidatalla voidaan tarkistaa opetusvaiheen mallin toimivuus. Käytännön sovellusten esimerkkeinä tästä voisi mainita asiakaskäyttäymisen ennustaminen ja tuotesuosittelu asiakkaalle. Ohjaamattomassa oppimisessa dataa ei ole luokiteltu, vaan se pyritään kuvaamaan oppimalla jokin dataan sopiva malli. Vahvistettu oppiminen perustuu ajatukseen, jossa ympäristö antaa positiivisen tai negatiivisen palautteen, ja algoritmi pyrkii löytämään ratkaisun joka antaa eniten posiitiivista palautetta.

Ohjatussa koneoppimisessa piirteiden erottaminen on keskeinen vaihe mallin kehittämisvaiheessa. Oikein löydetyillä piirteillä voi vaikuttaa algoritmin suorituskykyyn parantavasti, ja toisaalta huonosti valitut piirteet saavat aikaiseksi heikon algoritmin. Piirteet erotellaan datasta ennen opetusdatan käyttöä. Mallin suoritusta arvioidaan testidatalla ja tarvittaessa valitaan uusia piirteitä, ja testataan taas algoritmin toimivuutta.


- On ilmeistä, että ilman kunnon datamäärää järkevää analyysiä ei voida tehdä, joten analyysin teko alkaa  kattavan aineiston keräämisellä ja ylläpidolla.
- Nykyisin vallitsevana teknologiana käytetään tilastollista oppimista, jota on esimerkiksi koneoppiminen.
- Piirteiden erottamisella on merkittävä osuus algoritmin toimivuuden kannalta.
- Varsinaisia tekoälysysteemejä ei ole vielä kehitetty, tai julkaistu. 
- Lineaariregressio on yksinkertainen ohjatun oppimisen menetelmä. Data-alueen ulkopuolelta ennustaminen on epävarmempaa, sillä ei ole näyttöä siitä käyttäytyykö selitettävä muuttuja lineaarisesti tutkitun alueen ulkopuolella.

Tekoälyjärjestelmiä on olemassa monentasoisia, mutta yhdistävänä tekijänä on se että ne ovat kaikki ihmisen kehittelemiä ja ne vaativat paljon ihmistyötä toimiakseen.  Se synnyttää erilaisia rutiinitehtäviä, joita suorittavat työntekijät ympäri maailmaa. Dokumentti https://yle.fi/aihe/artikkeli/2020/04/23/ulkolinja-nakymattomat-tyontekijat-alipalkatut-avustajat-koulivat-tekoalya-ja kertoo alipalkatuista ja näkymättömäksi jääneistä internetin työntekijöistä, jotka omalta osaltaan antavat panoksensa yrityksille ja teknologialle halpaan hintaan.

Osallistuin opetukseen. Datana käytän kagglesta ladattua diabetes datasettiä, https://www.kaggle.com/tigganeha4/diabetes-dataset-2019.

In [ ]:

!pip install sklearn
!pip install numpy

In [ ]:
import pandas as pd
import numpy as np
from sklearn import preprocessing
from sklearn.impute import SimpleImputer

df=pd.read_csv('diabetes_dataset__2019.csv')
df.head()


Age Gender Family_Diabetes  ... Pdiabetes UriationFreq  Diabetic
0  50-59   Male              no  ...         0     not much        no
1  50-59   Male              no  ...         0     not much        no
2  40-49   Male              no  ...         0     not much        no
3  50-59   Male              no  ...         0     not much        no
4  40-49   Male              no  ...         0     not much        no

[5 rows x 18 columns]

Tarkastellaan sarakkeita Gender, Family_Diabetes ja Diabetic.

In [ ]:
data = df.loc[:,['Gender','Family_Diabetes','BMI','Diabetic']]
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 952 entries, 0 to 951
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Gender           952 non-null    object 
 1   Family_Diabetes  952 non-null    object 
 2   BMI              948 non-null    float64
 3   Diabetic         951 non-null    object 
dtypes: float64(1), object(3)
memory usage: 29.9+ KB


In [ ]:
data.isna().sum()

Gender             0
Family_Diabetes    0
BMI                4
Diabetic           1
dtype: int64

Diabetic-sarakkeessa on yksi puuttuva arvo, joten poistetaan tämä rivi.

In [ ]:
data = data.dropna(subset=['Diabetic'])

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 951 entries, 0 to 951
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Gender           951 non-null    object 
 1   Family_Diabetes  951 non-null    object 
 2   BMI              948 non-null    float64
dtypes: float64(1), object(2)
memory usage: 29.7+ KB


Muutetaan Diabetic-muuttujan arvot kategorisiksi arvoiksi.

In [ ]:
arr = preprocessing.LabelEncoder().fit_transform(data.Diabetic)
data.Diabetic = arr

In [ ]:
y = data.Diabetic

In [ ]:
data = data.drop(columns='Diabetic')


In [ ]:
data['BMI']

0      39.0
1      28.0
2      24.0
3      23.0
4      27.0
       ... 
947    25.0
948    27.0
949    23.0
950    27.0
951    30.0
Name: BMI, Length: 951, dtype: float64

In [ ]:
# Käytetään imputointimenetelmää puuttuviin BMI arvoihin korvaamalla ne havaittujen arvojen keskiarvolla.
#data['BMI'] = data['BMI'].astype(float)
imputer = SimpleImputer()


In [ ]:
data[['BMI']] = imputer.fit_transform(data[['BMI']])


data.isnull().sum()

Gender             0
Family_Diabetes    0
BMI                0
dtype: int64

In [ ]:


# Muutetaan kategoriset muuttuja dummy-arvoiksi

X = pd.get_dummies(data)
data.info()
X.info()
X.corr()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 951 entries, 0 to 951
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Gender           951 non-null    object 
 1   Family_Diabetes  951 non-null    object 
 2   BMI              951 non-null    float64
dtypes: float64(1), object(2)
memory usage: 29.7+ KB
<class 'pandas.core.frame.DataFrame'>
Int64Index: 951 entries, 0 to 951
Data columns (total 5 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   BMI                  951 non-null    float64
 1   Gender_Female        951 non-null    uint8  
 2   Gender_Male          951 non-null    uint8  
 3   Family_Diabetes_no   951 non-null    uint8  
 4   Family_Diabetes_yes  951 non-null    uint8  
dtypes: float64(1), uint8(4)
memory usage: 18.6 KB


BMI  ...  Family_Diabetes_yes
BMI                  1.000000  ...             0.114737
Gender_Female        0.119635  ...             0.096158
Gender_Male         -0.119635  ...            -0.096158
Family_Diabetes_no  -0.114737  ...            -1.000000
Family_Diabetes_yes  0.114737  ...             1.000000

[5 rows x 5 columns]

# Viikko 4

Tällä viikolla käydään läpi harjoitustyön prosessia.  [Harjoitustyön ohjeet](https://infotuni.github.io/joda2021/harjoitustyo/) löytyvät kurssin omilta sivuilta.

Datatiedeprojekti lähtee liikkeelle ongelmasta tai kysymyksestä johon halutaan vastaus. Tällä luennolla keskitytään ongelman tunnistamiseen, kenelle ratkaisua haetaan ja minkälaisia ratkaisuja on jo olemassa. Lisäksi voi pohtia, minkälaisia ratkaisuja projekti tarjoaa, miten ratkaisua käytetään ja miten käyttäjät hyötyvät siitä, miten ratkaisun arvoa voi mitata, miten se vaikuttaa tulevaan toimintaan ja mitä koko projekti maksaa.

Luennolla käsiteltiin  Airbnb:n majoitus- ja elämysjohtajan viikkokatsausta. Liiketoimintaongelmana on ilmennyt, että yritys on saanut negatiivista palautetta sekä vieraskäyttäytymisestä että majoituskohteista. Arviointipalautteista tehtyjen analyysien pohjalta on päädytty toimenpiteisiin, joilla lisätään palvelun laatua ja luotettavuutta.

Miten mitata data-analyysin hyödyllisyyttä? Tähän on esitetty kolme kohtaa: laatu, nopeus ja robustisuus. Laadun indikaattorina voidaan käyttää esimerkiksi tuottoa, hintaa, tehokkuutta tai asiakastyytyväisyyttä. Nopeus voi auttaa saavuttamaan paremman aseman markkinoilla ja ainakin työmäärä kevenee. Hyvässä mallissa otetaan huomioon myös poikkeukset ja vaihtelevat tilanteet jotta malli on kestävä vaikka olosuhteet muuttuisivat.

Miten aloittaa harjoitustyö? Todo-lista:
- Mieti jokin kiinnostava kysymys johon haluat vastauksen.
- Mieti onko siihen olemassa sopivaa dataa ja millä tavoin hankit sen.
- Tutki dataa; minkälaista tietoa se sisältää? Aloita datan putsaaminen: Tarkista virheet ja tietotyypit, tarkista muuttujat ja puuttuvat arvot.
- Rikasta dataa. Löydä keino yhdistää eri lähteistä tulevat data yhdeksi. Muuta muuttujia muotoon jotka palvelevat ongelmakysymystä paremmin.
-Visualisointi: mieti mitkä kuvat ovat oikeasti arvokkaita. Katso vinkit jinja2, tablesorter, streamlit.
- Aloita mallintaminen.


Alustuksena voisi luennon alkuun käydä läpi vielä tarkemmin CRISP-DM -prosessia, mitä sillä tarkoitetaan, mikä on sen rakenne. Silloin olisi helpompi seurata esimerkkiäkin prosessin näkökulmasta.

Ajatusten vaihtoa voisi heitellä myös koko ryhmän kokoisena kuin erillisissä breakout huoneissa. Kaikki eivät kumminkaan ole niin innokkaita osallistumaan, joten tehtävän alustus ja esimerkkien anto voisi antaa paremmin kuvan mitä tehtävissä haetaan ja saada ajatustenvaihtoa lennokkaammaksi.

Jatkan edellisen viikon kagglesta ladatun https://www.kaggle.com/tigganeha4/diabetes-dataset-2019 diabetes-datasetin käsittelyä. Data on jo siivottu, joten jaetaan aineisto opetus- ja testiaineistoon.

In [ ]:
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error

# Viikko 5

https://github.com/TeemuMikkonen/JODA-NLP

https://data.solita.fi/finnish-stemming-and-lemmatization-in-python/

